src/tools/compiletest/src/runtest.rs

```diff
@@ -3339,13 +3347,16 @@ impl<'test> TestCx<'test> {
         if self.props.run_rustfix && self.config.compare_mode.is_none() {
             // And finally, compile the fixed code and make sure it both
             // succeeds and has no diagnostics.
-            let rustc = self.make_compile_args(
-                &self.testpaths.file.with_extension(UI_FIXED),
+            let mut rustc = self.make_compile_args(
+                &self.expected_output_path(UI_FIXED),
                 TargetLocation::ThisFile(self.make_exe_name()),
                 emit_metadata,
                 AllowUnused::No,
                 LinkToAux::Yes,
             );
+            // Set the crate name to avoid `file.revision.fixed` inferring the
+            // invalid name `file.revision`
+            rustc.arg("--crate-name=fixed");
```

Comment:
Currently files with both revisions and `// run-rustfix`  check `test.fixed` instead of `test.revision.fixed`, that gets fixed just above here

However `rustc` seems to use `file_stem` of the path to determine a default crate name if you don't provide your own, so it infers `test.revision` here, but considers that invalid:

```
$ rustc test.revision.fixed                 
error: invalid character `'.'` in crate name: `test.revision`

error: aborting due to previous error
```

Reply:
There don't seem to be any revision + `// run-rustfix` tests in the rust suite, but there are some leftovers from a previous one that's no longer used so I'll `rm` those